File: src/Module/User/UserEntity.java
package Module.User;

import Module.File.FileEntity;

import java.io.Serializable;
import java.util.List;
import java.util.stream.Collectors;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class UserEntity implements Serializable {
    public int userId;
    public String userName;
    public String passWord;
    private List<FileEntity> fileEntityList;

    public UserEntity() {
    }

    public UserEntity(int userId, String userName, String passWord) {
        this.userId = userId;
        this.userName = userName;
        this.passWord = passWord;
    }

    public UserEntity(UserModel UserModel) {
        this.userId = UserModel.getUserId();
        this.userName = UserModel.getUserName();
        this.passWord = UserModel.getPassWord();
        if (UserModel.getFilesByUserId() != null)
            this.fileEntityList = UserModel.getFilesByUserId().parallelStream().map(FileEntity::new).collect(Collectors.toList());
           }

    public UserModel toEntity() {
        UserModel UserModel = new UserModel();
        UserModel.setUserId(userId);
        UserModel.setUserName(userName);
        UserModel.setPassWord(passWord);
        if (fileEntityList != null)
            UserModel.setFilesByUserId(fileEntityList.parallelStream().map(FileEntity::toEntity).collect(Collectors.toList()));
             return UserModel;
    }
}

File: web/Angular4/src/app/Modules/user/user.Entity.ts
import {FileEntity} from "../file/file.Entity";

export class UserEntity {
    userId: number;
    userName: string;
    passWord: string;
    fileEntities: FileEntity[];
    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;

    constructor(user: any = null) {
        if (user == null) {
            this.userId = null;
            this.userName = null;
            this.passWord = null;
            this.fileEntities = [];
        } else {
            this.userId = user.userId;
            this.userName = user.userName;
            this.passWord = user.passWord;
            if (user.fileEntities != null) {
                this.fileEntities = [];
                for (let item of user.fileEntities) {
                    this.fileEntities.push(new FileEntity(item));
                }
            }

            if (this.fileEntities == null) this.fileEntities = [];
        }
        this.IsEdit = false;
    }
}